tutorials/getting_started/creating_a_model.md

```diff
@@ -0,0 +1,187 @@
+   for          IN   I-PP  O
+   Baghdad      NNP  I-NP  I-LOC
+   .            .    O     O
+```
+
+where each line contains a token, a part-of-speech tag, a syntactic chunk tag, and a named-entity tag.
+An empty line indicates the end of a sentence, and a line
+
+```
+-DOCSTART- -X- O O
+```
+
+indicates the end of a document. (Our reader is concerned only with sentences
+and doesn't care about documents.)
+
+You can poke at the code yourself, but at a high level we use
+[`itertools.groupby`](https://docs.python.org/3/library/itertools.html#itertools.groupby)
+to chunk our input into groups of either "dividers" or "sentences".
+Then for each sentence we split each row into four columns,
+create a `TextField` for the token, and create a `SequenceLabelField`
+for the tags (which for us will be the NER tags).
+
+## Creating a Config File
+
+As the `CrfTagger` model is quite similar to the `SimpleTagger` model,
+we can get away with a similar configuration file. We need to make only
+a couple of changes:
+
+* change the `model.type` to `"crf_tagger"`
+* change the `"dataset_reader.type"` to `"conll2003"`
+* add a `"dataset_reader.tag_label"` field with value "ner" (to indicate that the NER labels are what we're predicting)
+
+We don't *need* to, but we also make a few other changes
+
+* following [Peters, Ammar, Bhagavatula, and Power 2017](https://www.semanticscholar.org/paper/Semi-supervised-sequence-tagging-with-bidirectiona-Peters-Ammar/73e59cb556351961d1bdd4ab68cbbefc5662a9fc), we use a GRU character encoding
+as well as a GRU for our stacked encoder
+* we also start with pretrained GloVe vectors for our token embeddings
+* we add a regularizer that applies a L2 penalty just to the `transitions`
+  parameters to help avoid overfitting
+* we add a `test_data_path` and set `evaluate_on_test` to true.
+  This is mostly to ensure that our token embedding layer loads the GloVe
+  vectors corresponding to tokens in the test data set, so that they are not
+  treated as out-of-vocabulary at evaluation time. The second flag just evaluates
+  the model on the test set when training stops. Use this flag cautiously,
+  when you're doing real science you don't want to evaluate on your test set too often.
+
+## Putting It All Together
+
+At this point we're ready to train the model.
+In this case our new classes are part of the `allennlp` library,
+which means we can just use `allennlp/run.py train`,
+but if you were to create your own model they wouldn't be.
+
+In that case `allennlp/run.py` never loads the modules in which
+you've defined your classes, they never get registered, and then
+AllenNLP is unable to instantiate them based on the configuration file.
+
+In such a case you'll need to create your own such script.
+You can actually copy that one, the only change you need to make
+is to import all of your custom classes at the top:
+
+```python
+from myallennlp.models import CrfTagger
+from myallennlp.modules import ConditionalRandomField
```

Comment:
The CRF isn't created in the tagger from the registry, so it's not necessary here.  But the dataset reader is.

Reply:
fixed